src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public partial class DependencyContextJsonReader : IDependencyContextReader
+    {
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            DependencyContext dependencyContext;
+            try
+            {
+                dependencyContext = ReadCore(drained.AsSpan(0, drained.Count));
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+                throw;
+            }
+            finally
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+            }
+            return dependencyContext;
+        }
+
+        private static bool IsTokenTypeProperty(JsonTokenType tokenType)
+            => tokenType == JsonTokenType.PropertyName;
+
+        private DependencyContext ReadCore(ReadOnlySpan<byte> jsonData)
+        {
+            var reader = new Utf8JsonReader(jsonData, isFinalBlock: true, state: default);
+
+            reader.ReadStartObject();
+
+            string runtime = string.Empty;
+            string framework = string.Empty;
+            bool isPortable = true;
+            string runtimeTargetName = null;
+            string runtimeSignature = null;
+
+            CompilationOptions compilationOptions = null;
+            List<Target> targets = null;
+            Dictionary<string, LibraryStub> libraryStubs = null;
+            List<RuntimeFallbacks> runtimeFallbacks = null;
+
+            while (reader.Read() && IsTokenTypeProperty(reader.TokenType))
+            {
+                switch (reader.GetString())
+                {
+                    case DependencyContextStrings.RuntimeTargetPropertyName:
+                        ReadRuntimeTarget(ref reader, out runtimeTargetName, out runtimeSignature);
+                        break;
+                    case DependencyContextStrings.CompilationOptionsPropertName:
+                        compilationOptions = ReadCompilationOptions(ref reader);
+                        break;
+                    case DependencyContextStrings.TargetsPropertyName:
+                        targets = ReadTargets(ref reader);
```

Comment:
I'm still a bit concerned about the amount of code that would need to be updated in two places if we need to keep supporting two different readers.

If it were me, I'd revert DependencyContextJsonReader, then add a new `internal ref struct UnifiedJsonReader` which provides the interface that this type wants (`ReadStartObject()`, et cetera) and has one implementation using Utf8JsonReader and another using the NewtonSoft one.  (And then update all the methods to take the `ref UnifiedJsonReader`, of course)

Reply:
I am not sure how much we should invest in this `UnifiedJsonReader` approach if we can drop the other TFMs and the dependency on Newtonsoft.Json all together.

Do you think I should revert and unify even if we will remove one of the impls soon? In other words, it is acceptable to support two readers in the short term while we evaluate the TFMs required and then consolidate only if we decide to keep the `JsonTextReader` impl as well?

Reply:
> Do you think I should revert and unify even if we will remove one of the impls soon?

Yep.  If the older refs get dropped you just stop building the Newtonsoft-using version.  The shim type performs the same role as the current extension methods.

And if we stay in limbo a long time (or decide we need to keep building the older TFMs), it keeps the limbo state cleaner.